packages/flutter/lib/src/material/scaffold.dart

```diff
@@ -37,10 +40,208 @@ enum _ScaffoldSlot {
   statusBar,
 }
 
-/// Geometry information for scaffold components.
+/// Responsible for placing the [FloatingActionButton] in the [Scaffold].
+/// 
+/// Flutter provides [FabPositioner]s for the common FAB placements in
+/// Material Design apps, whose implementations are available as static
+/// members of this class.
+abstract class FabPositioner {
+  const FabPositioner();
```

Comment:
please use the following boilerplate for the docs of const abstract constructors:
```
  /// Abstract const constructor. This constructor enables subclasses to provide
  /// const constructors so that they can be used in const expressions.
```

Reply:
(same comment in other places)

Reply:
Done